### PR TITLE
Dedupe BigNum util

### LINKED ISSUE #2751: De-dupe BigNumber related utils
We sometimes use ethers v5 BigNumber and other times use BigNumber.js (which is better). There's also an option to use neither and rely on native BigInt support.
See comment here: https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/2744#discussion_r1333572083

**Requirements:**
1. All public facing interfaces / methods use `string` typed-numbers, because they're universal and agnostic to any particular library.
2. In implementations, we always use BigNumber.js to parse number strings and do arithmetic
3. Create utilities for converting between BigNumber.js and Ethers-v5-BigNumber
4. Use either strings or Ethers-v5-BigNumber when interacting with Ethers contract (as they won't accept BigNumber.js). Whichever is more convenient.

### PR DESCRIPTION
**Pull request by ottbunn**
### Description
wip PR for de-dupe BigNumber libraries usage, having some discussion in the issue tab

### Related issues

- Fixes #2751

### CODE REVIEW

File: typescript/utils/src/amount.ts [hunk 1]
```diff
@@ -96,50 +119,26 @@ export function eqAmountApproximate(
...
-    const difference = fromDecimals - toDecimals;
-    return value.div(ethers.BigNumber.from('10').pow(difference));
+      .integerValue(BigNumber.ROUND_FLOOR)
+      .toString();
   }
   // fromDecimals < toDecimals
   else {
     const difference = toDecimals - fromDecimals;
-    return value.mul(ethers.BigNumber.from('10').pow(difference));
+    return amount.times(new BigNumber(10).pow(difference)).toString();
```
Comment (jmrossy) [inline typescript/utils/src/amount.ts:142]: Nit, you probably don't need to wrap `10` in a BN here

File: typescript/utils/src/big-numbers.test.ts [hunk 2]
```diff
@@ -1,12 +1,65 @@
...
+import {
+  bigToFixed,
+  fixedToBig,
+  isBigNumberish,
+  isZeroish,
+  mulBigAndFixed,
+} from './big-numbers';
 
 describe('utils', () => {
+  describe('isBigNumberish', () => {
```
Comment (jmrossy) [inline typescript/utils/src/big-numbers.test.ts:14]: Very nice, thanks for these new tests

**Review by jmrossy — COMMENTED** [review]
So far this is looking great!

**Comment by jmrossy** [comment]
@ottbunn it looks like there are places outside the utils package that are broken by these changes. Do you have permission to see the logs of the failing CI tests? 

I don't think you made any mistakes. I think we just need additional changes. Feel free to look into the errors yourself, otherwise I'll pick up this changeset soon and finish it on my end.

**Comment by ottbunn** [comment]
ah oops, I'll try to fix them today 🙏🏻

**Comment by ottbunn** [comment]
updated, umm looks like I'll need your help to trigger the CI (?) @jmrossy

**Comment by jmrossy** [comment]
Thanks @ottbunn, sorry for the delay in getting this merged.
I'm going to rebase this against our v3 branch, which is our new working branch that will soon be merged to main. Planning to merge this today :)

**Review by jmrossy — APPROVED** [review]

**Pull request merged**